app/questionnaire/questionnaire_store_updater.py

```diff
@@ -298,16 +327,39 @@ def update_answers(
 
             answer_updated = self._update_answer(answer_id, list_item_id, answer_value)
             if answer_updated:
-                self._capture_dependencies_for_answer(answer_id)
+                self._capture_section_dependencies_for_answer(answer_id)
+                self._capture_block_dependencies_for_answer(answer_id)
 
-    def update_progress_for_dependant_sections(self) -> None:
+    def update_progress_for_dependent_sections(self) -> None:
         """Removes dependent blocks from the progress store and updates the progress to IN_PROGRESS.
+        Section progress is not updated for the current location as it is handled by `handle_post` on block handlers."""
 
-        Section progress is not updated for the current location as it is handled by `handle_post` on block handlers.
+        self._remove_dependent_blocks_and_capture_dependent_sections()
+
+        for section in self.dependent_sections:
+            is_path_complete = section.is_complete
+            is_started = (
+                self._progress_store.get_section_status(
+                    section.section_id, section.list_id
+                )
+                != "NOT_STARTED"
+            )
+
+            if section.is_complete is None and is_started:
```

Comment:
We should do `is_started` logic inline, as you only need to do that is `section.is_complete` is None. If it was True/False, then the section had to have started.

Reply:
Yeah I read your mind and put it inline, before even reading this comment!